PLATER/services/util/question.py

```diff
@@ -3,53 +3,96 @@
-    KNOWLEDGE_GRAPH_KEY='knowledge_graph'
-    NODES_LIST_KEY='nodes'
-    EDGES_LIST_KEY='edges'
-    NODE_TYPE_KEY='category'
-    EDGE_TYPE_KEY='predicate'
-    SOURCE_KEY='subject'
-    TARGET_KEY='object'
-    NODE_BINDINGS_KEY='node_bindings'
-    EDGE_BINDINGS_KEY='edge_bindings'
+class Question:
+    # SPEC VARS
+    QUERY_GRAPH_KEY = 'query_graph'
+    KG_ID_KEY = 'ids'
+    QG_ID_KEY = 'ids'
+    ANSWERS_KEY = 'results'
+    KNOWLEDGE_GRAPH_KEY = 'knowledge_graph'
+    NODES_LIST_KEY = 'nodes'
+    EDGES_LIST_KEY = 'edges'
+    NODE_TYPE_KEY = 'categories'
+    EDGE_TYPE_KEY = 'predicate'
+    SOURCE_KEY = 'subject'
+    TARGET_KEY = 'object'
+    NODE_BINDINGS_KEY = 'node_bindings'
+    EDGE_BINDINGS_KEY = 'edge_bindings'
     CURIE_KEY = 'curie'
 
     def __init__(self, question_json):
         self._question_json = copy.deepcopy(question_json)
+        self.bl_url = f'https://raw.githubusercontent.com/biolink/biolink-model/1.8.2/biolink-model.yaml'
+        self.toolkit = Toolkit(self.bl_url)
+        self.provenance = os.environ.get('PROVENANCE_TAG', 'infores:automat.unknown')
 
     def compile_cypher(self):
         return get_query(self._question_json[Question.QUERY_GRAPH_KEY])
 
-    @staticmethod
-    def format_attribute_trapi_1_1 (kg_items):
+    # @staticmethod
+    def format_attribute_trapi_1_1(self, kg_items):
         for identifier in kg_items:
-            item = kg_items[identifier]
-            attributes = item.get('attributes', {})
-            for attr in attributes:
+            # get the properties for the record
+            props = kg_items[identifier]
+
+            # save the transpiler attribs
+            attributes = props.get('attributes', [])
+
+            # create a new list that doesnt have the core properties
+            new_attribs = [attrib for attrib in attributes if attrib['original_attribute_name'] not in props]
+
+            # for the non-core properties
+            for attr in new_attribs:
+                # make sure the original_attribute_name has somthig other than none
                 attr['original_attribute_name'] = attr['original_attribute_name'] or ''
-                # uses Data as attribute type id if not defined
-                if not('attribute_type_id' in attr and attr['attribute_type_id'] != 'NA'):
-                    attr['attribute_type_id'] = 'EDAM:data_0006'
-                if not('value_type_id' in attr and attr['value_type_id'] != 'NA'):
-                    # fix for issue https://github.com/RENCI-AUTOMAT/Automat-server/issues/15
-                    attr['value_type_id'] = 'biolink:Attribute'
-                if attr['attribute_type_id'] == "biolink:original_knowledge_source":
-                    attr['value_type_id'] = 'biolink:InformationResource'
+
+                # lookup the biolink info
+                bl_info = self.toolkit.get_element(attr['original_attribute_name'])
+
+                # did we get something
+                if bl_info is not None:
```

Comment:
Hi @PhillipsOwen , when testing this against a neo4j instance, the instance has some nodes that have gene nodes with attribute gene_family and so this block of code executes. bl_info comes back alright with a lookup for 'gene_family' as the original_attribute_name . But when it does bl_info['slot_uri'] it breaks with key not found error. The whole bl_info object looks like 
```
ClassDefinition(
name='gene family', id_prefixes=['PANTHER.FAMILY', 'HGNC.FAMILY', 'FB', 'interpro', 'CATH', 'CDD', 'HAMAP', 'PFAM', 'PIRSF', 'PRINTS', 'PRODOM', 'PROSITE', 'SMART', 'SUPFAM', 'TIGRFAM', 'CATH.SUPERFAMILY', 'RFAM', 'KEGG.ORTHOLOGY', 'EGGNOG'], 
definition_uri='https://w3id.org/biolink/vocab/GeneFamily', 
aliases=['orthogroup', 'protein family'], 
local_names={}, 
mappings=[], 
description='any grouping of multiple genes or gene products related by common descent', 
alt_descriptions={}, 
deprecated=None,
 todos=[], 
 notes=[], 
 comments=[], 
 examples=[], 
 in_subset=['model_organism_database'], 
 from_schema='https://w3id.org/biolink/biolink-model', 
 imported_from=None, 
 see_also=[], 
 exact_mappings=['NCIT:C26004', 'WIKIDATA:Q2278983'], 
 close_mappings=[], 
 related_mappings=[], 
 narrow_mappings=['SIO:001380', 'NCIT:C20130', 'WIKIDATA:Q417841'], 
 broad_mappings=[], 
 deprecated_element_has_exact_replacement=None, 
 deprecated_element_has_possible_replacement=None, 
 extensions={}, 
 annotations={}, 
 is_a='molecular entity', 
 abstract=None, 
 mixin=None, 
 mixins=['gene grouping mixin'], apply_to=[], values_from=[], created_by=None, created_on=None, last_updated_on=None, modified_by=None, status=None, string_serialization=None, slots=['id', 'iri', 'type', 'name', 'description', 'source', 'provided by', 'has attribute', 'named thing_category', 'in taxon', 'has gene or gene product'], slot_usage={}, attributes={}, class_uri='biolink:GeneFamily', subclass_of=None, union_of=[], defining_slots=[], tree_root=None)
```
In there i was able to see class_uri but not slot_uri maybe those are the same thing and a new change to bl 1.8.2?

Reply:
And also testing this with class_uri some bl_info (concepts) don't have range